internal/dataplane/kongstate/kongstate_test.go

```diff
@@ -335,64 +346,226 @@ func TestFillConsumersAndCredentials(t *testing.T) {
 				"hash_secret":   []byte("true"),
 			},
 		},
-	}
-	consumers := []*configurationv1.KongConsumer{
 		{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      "foo",
+				Name:      "emptyCredSecret",
+				Namespace: "default",
+			},
+			Data: map[string][]byte{
+				"kongCredType": []byte("key-auth"),
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "unsupportedCredSecret",
 				Namespace: "default",
-				Annotations: map[string]string{
-					"kubernetes.io/ingress.class": annotations.DefaultIngressClass,
-				},
 			},
-			Username: "foo",
-			CustomID: "foo",
-			Credentials: []string{
-				"fooCredSecret",
-				"barCredSecret",
+			Data: map[string][]byte{
+				"kongCredType": []byte("unsupported"),
+				"foo":          []byte("bar"),
 			},
 		},
 	}
-	store, _ := store.NewFakeStore(store.FakeObjects{
-		Secrets:       secrets,
-		KongConsumers: consumers,
-	})
 
-	want := KongState{
-		Consumers: []Consumer{{
-			Consumer: kong.Consumer{
-				Username: kong.String("foo"),
-				CustomID: kong.String("foo"),
-			},
-			KeyAuths: []*KeyAuth{{kong.KeyAuth{
-				Key: kong.String("whatever"),
-				TTL: kong.Int(1024),
-			}}},
-			Oauth2Creds: []*Oauth2Credential{
+	testCases := []struct {
+		name                        string
+		k8sConsumers                []*configurationv1.KongConsumer
+		expectedKongStateConsumers  []Consumer
+		expectedTranslationFailures map[k8stypes.NamespacedName]failures.ResourceFailure
```

Comment:
As I can see we're only using `ResourceFailure.Message()` in the assertions - how about changing this like that:
```suggestion
		expectedTranslationFailuresMessages map[k8stypes.NamespacedName]string
```

Reply:
done.